docs/DeveloperGuide.md

```diff
@@ -4,26 +4,207 @@
+Timings that are not marked 'busy' are then identified as free time slots.
+
+The following sequence diagram shows how the command `free` is executed.
+![FreeCommandSequenceDiagram](diagrams/FreeCommandSequenceDiagram.png)
+
+## 3.4 Delete events `delete`
+Deletes an event from the user's specified timetable
+
+**Format:** ` delete n/NAME i/INDEX`
+
+* Deletes from the timetable of user
+* Deletes the lesson at the specified `INDEX`.
+* The `INDEX` refers to the index number shown in the displayed lesson list.
+
+Example of usage:
+
+`delete n/John i/1`
+
+If the event is a meeting, it will delete this event from all users
+
+The following sequence diagram shows how the `delete` command works:
+![DeleteCommandSequenceDiagram](diagrams/DeleteCommandSequenceDiagram.png)
+
+Before the timetable is listed out, it will also be sorted according to day and time for easy reading.
+
+## 3.4 Data Saving Feature
+
+The save mechanism is facilitated by StorageFile. It creates a text file in the local machine and writes all the elements
+in the MasterTimetable and Timetable to the text file in human-readable format.
+
+Given below is the steps on how the save operation works.
+
+Step 1: When the user exits the application. The StorageFile.saveData will be initialized.
+
+Step 2: It will check for the existence of the data file. If the file is not found, it will create the file.
+Else do nothing.
+
+Step 3: After creating the file, collateAll method will be called to consolidate all elements in the MasterTimetable and Timetable list,
+and write them into the data file.
+
+Step 4: A message will be shown to the user, informing him that the data has been saved successfully. For any exception
+encountered along the way, an error message will be shown.
+
+The following sequence diagram shows how the save operation works:
+![SaveSequenceDiagram](diagrams/SaveSequenceDiagram.png)
+
+## 3.5 Loading Data Feature
+
+The save mechanism is facilitated by StorageFile. It reads the data specified in the local machine and stores all 
+the records back into the MasterTimetable and Timetable list.
+
+Given below is the steps on how the load operation works.
+
+Step 1: When the user starts the application. The StorageFile.LoadData will be initialized.
+
+Step 2: It will check for the existence of the data file. If the file is not found, do nothing.
+Else it will open the file and extract the record line by line.
+
+Step 3: If a record is a lesson type, then it will call ParserLocalData.prepareLoadLesson method.
+Else if the record is a meeting type, then it will add the record into the meetingList.
+
+Step 4: If there is no record left, the program will check if the previous record contain a meeting.
+If yes, it will call for ParserLocalData.prepareLoadMeeting method to add all the meetings into the list.
+
+The following sequence diagram shows how the load operation works:
+![LoadSequenceDiagram](diagrams/LoadSequenceDiagram.png)
```

Comment:
Could you consider splitting the different steps in Storage into multiple sequence diagrams?

Use ref frames to break up the content, e.g.

The bottom part (where you load meetings) could be a ref frame, which you can elaborate on later

![image](https://user-images.githubusercontent.com/26570382/161183282-5aee2f5f-6df4-433e-b5ce-0b6f47db4ab8.png)

Reply:
Since theres only one condition, can the 2nd alt be replaced by opt?